components/ruby/lib/license_acceptance/strategy/file.rb

```diff
@@ -29,11 +29,18 @@ def accepted?(product_relationship)
 
         searching.each do |product|
           config.license_locations.each do |loc|
-            f = ::File.join(loc, product.filename)
-            if ::File.exist?(f)
-              logger.debug("Found license #{product.filename} at #{f}")
-              missing_licenses.delete(product)
-              break
+            license_path = ::File.join(loc, product.filename)
+            if ::File.exist?(license_path)
+              ::File.open(license_path, ::File::RDONLY) do |license_file|
+                license = YAML.load(license_file.read)
+                if product.license.name == license['license_name']
```

Comment:
If users have previously accepted a license for (IE) Chef Infra, we do not need to re-prompt them for acceptance, even if their file does not include the `license_name`.

I think we need to assume they accepted `'EULA'` if `license_name` is not present.

Reply:
👍 Implemented in 4b87615